pints/_plot.py

```diff
@@ -0,0 +1,248 @@
+    `problem`
+        pints.SingleSeriesProblem object
+
+    """
+    if not isinstance(problem, pints.SingleSeriesProblem):
+        raise TypeError('pints.SingleSeriesProblem type is expected')
+
+    n_sample, n_param = chain.shape
+    plot_n_sample = 1000 if n_sample>1000 else n_sample
+    times = problem.times()
+
+    # Evaluate the model for all inferred parameters
+    predicted_values = []
+    for params in chain[-plot_n_sample:,:]:
+        predicted_values.append(problem.evaluate(params[:-1]))
+    predicted_values = np.array(predicted_values)
+    mean_values = np.mean(predicted_values, axis=0)
+
+    # Plot prediction
+    fig = plt.figure(figsize=(7.5, 3.75))
+    plt.xlabel('Time')
+    plt.ylabel('Value')
+    plt.plot(times, predicted_values[0], color='#1f77b4', label='inferred series')
+    for v in predicted_values[1:]:
+        plt.plot(times, v, color='#1f77b4', alpha=0.05)
+    plt.plot(times, mean_values, color='black', lw=2, label='mean inferred')
+    plt.plot(times, problem.values(), 'o', color='#7f7f7f', ms=6.5, label='data points')
+
+    return fig
+
+def forceAspect(ax, aspect=1):
+    """
+    Set aspect.
+
+    `ax`
+        matplotlib axes handler
+    `aspect`
+        aspect to be set
+
+    """
+    im = ax.get_images()
+    ex = im[0].get_extent()
+    ax.set_aspect(abs((ex[1]-ex[0])/(ex[3]-ex[2]))/aspect)
+
+def hist2d(X, Y, ax):
+    """
+    2D histogram plot.
+
+    `X`
+        An array of the first variable
+    `Y`
+        An array of the second variable
+    `ax`
+        matplotlib axes handler
+
+    """
+    Xmin, Xmax = np.min(X), np.max(X)
+    Ymin, Ymax = np.min(Y), np.max(Y)
+    x2 = np.linspace(Xmin, Xmax, 25)
+    y2 = np.linspace(Ymin, Ymax, 25)
+    histplot = ax.hist2d(X, Y, bins=[x2,y2], normed=True, cmap=plt.cm.Blues)
+    forceAspect(ax)
+
+def kde2d(X, Y, ax):
+    """
+    2D kernel density estimation plot.
+
+    `X`
+        An array of the first variable
+    `Y`
+        An array of the second variable
+    `ax`
+        matplotlib axes handler
+
+    """
+    Xmin, Xmax = np.min(X), np.max(X)
+    Ymin, Ymax = np.min(Y), np.max(Y)
+    X1, Y1 = np.mgrid[Xmin:Xmax:100j, Ymin:Ymax:100j]
+    positions = np.vstack([X1.ravel(), Y1.ravel()])
+    values = np.vstack([X, Y])
+    kernel = stats.gaussian_kde(values)
+    Z = np.reshape(kernel(positions).T, X1.shape)
+    ax.imshow(np.rot90(Z), cmap=plt.cm.Blues, extent=[Xmin, Xmax, Ymin, Ymax])
+    ax.plot(X, Y, 'k.', markersize=2)
+    forceAspect(ax)
+
+def kde1d(X, ax):
+    """
+    1D kernel density estimation plot.
+
+    `X`
+        An array of the variable
+    `ax`
+        matplotlib axes handler
+
+    """
+    Xmin = np.min(X)
+    Xmax = np.max(X)
+    x1 = np.linspace(Xmin, Xmax, 100)
+    x2 = np.linspace(Xmin, Xmax, 50)
+    kernel = stats.gaussian_kde(X)
+    Z = kernel(x1)
+    ax.hist(X, bins=x2, normed=True)
+    ax.plot(x1, Z)
+    #forceAspect(ax)
+
+def pairwise_hist(chain):
```

Comment:
I wouldn't use all these abbreviations in the method names! Just make it `pairwise_histogram` full out (or `pairwise_scatterplot`)